dozer/cogs/roles.py

```diff
@@ -0,0 +1,178 @@
+			e.add_field(name='You already have {} role(s)!'.format(len(already_have)), value='\n'.join(already_have_names), inline=False)
+		extra = len(norm_names) - len(valid)
+		if extra > 0:
+			e.add_field(name='{} role(s) could not be found!'.format(extra), value='Use `{0.prefix}{0.invoked_with} list` to find valid giveable roles!'.format(ctx), inline=False)
+		await ctx.send(embed=e)
+	
+	@giveme.command()
+	@bot_has_permissions(manage_roles=True)
+	@has_permissions(manage_guild=True)
+	async def add(self, ctx, *, name):
+		"""Makes an existing role giveable, or creates one if it doesn't exist. Name must not contain commas.
+		Similar to create, but will use an existing role if one exists."""
+		if ',' in name:
+			raise BadArgument('giveable role names must not contain commas!')
+		norm_name = self.normalize(name)
+		with db.Session() as session:
+			settings = session.query(GuildSettings).filter_by(id=ctx.guild.id).one_or_none()
+			if settings is None:
+				settings = GuildSettings(id=ctx.guild.id)
+				session.add(settings)
+			if norm_name in (giveable.norm_name for giveable in settings.giveable_roles):
+				raise BadArgument('that role already exists and is giveable!')
+			candidates = [role for role in ctx.guild.roles if self.normalize(role.name) == norm_name]
+			
+			if not candidates:
+				role = await ctx.guild.create_role(name=name, reason='Giveable role created by {}'.format(ctx.author))
+			elif len(candidates) == 1:
+				role = candidates[0]
+			else:
+				raise BadArgument('{} roles with that name exist!'.format(len(candidates)))
+			settings.giveable_roles.append(GiveableRole.from_role(role))
+		await ctx.send('Role "{0}" added! Use `{1}{2} {0}` to get it!'.format(role.name, ctx.prefix, ctx.command.parent))
+	
+	@giveme.command()
+	@bot_has_permissions(manage_roles=True)
+	@has_permissions(manage_guild=True)
+	async def create(self, ctx, *, name):
+		"""Create a giveable role. Name must not contain commas.
+		Similar to add, but will always create a new role."""
+		if ',' in name:
+			raise BadArgument('giveable role names must not contain commas!')
+		norm_name = self.normalize(name)
+		with db.Session() as session:
+			settings = session.query(GuildSettings).filter_by(id=ctx.guild.id).one_or_none()
+			if settings is None:
+				settings = GuildSettings(id=ctx.guild.id)
+				session.add(settings)
+			if norm_name in (giveable.norm_name for giveable in settings.giveable_roles):
+				raise BadArgument('that role already exists and is giveable!')
+			
+			role = await ctx.guild.create_role(name=name, reason='Giveable role created by {}'.format(ctx.author))
+			settings.giveable_roles.append(GiveableRole.from_role(role))
+		await ctx.send('Role "{0}" created! Use `{1}{2} {0}` to get it!'.format(role.name, ctx.prefix, ctx.command.parent))
+	
+	@giveme.command()
+	@bot_has_permissions(manage_roles=True)
+	async def remove(self, ctx, *, roles):
+		"""Removes multiple giveable roles from you. Names must be separated by commas."""
```

Comment:
you should add that it can contain spaces in the separator as well

Reply:
I asked a sample of potential users for an example of a comma-separated list, and their responses include spaces. I believe it's common knowledge. Anybody else want to chime in?

Reply:
Explicit is better than implicit